crates/matrix-sdk/src/event_cache/room/events.rs

```diff
@@ -398,6 +422,20 @@ impl RoomEvents {
     }
 }
 
+pub(in crate::event_cache) struct AddEventReport {
+    /// Number of new unique events that have been added.
+    num_new_unique: usize,
+    /// Number of events which have been deduplicated.
+    num_duplicated: usize,
+}
+
+impl AddEventReport {
+    /// Were all the events (at least one) we added already known?
+    pub fn deduplicated_all_new_events(&self) -> bool {
+        self.num_new_unique > 0 && self.num_new_unique == self.num_duplicated
+    }
```

Comment:
Can't you simply test `self.num_duplicated > 0`?

Actually, I don't understand the code here. The code, the doc, and the method name don't seem to do the same thing. Can you revisit this a bit please?

Reply:
I've rewritten the last commit and put a super large comment, hopefully it helps. Let me know if it doesn't, post-merge, and we can rephrase it :pray:

Reply:
I like you've removed `AddEventReport`, it seems cleaner this way. I may have prefer a `Option<T>` instead of `(bool, T)`, but it's a personal taste (I believe it matches `std` a bit better, but that's a detail, I'll try to submit a PR to not forget that and being able to discuss it).